src/lib/Microsoft.Health.Fhir.R4.Ingest/Host/FhirHealthCheckExtensions.cs

```diff
@@ -33,8 +36,17 @@ internal static IWebJobsBuilder AddFhirHealthCheck(this IWebJobsBuilder builder)
             builder.Services.Configure<FhirClientFactoryOptions>(config.GetSection("FhirClient"));
 
             // Register services
-            builder.Services.TryAddSingleton<IFactory<FhirClient>, FhirClientFactory>();
-            builder.Services.TryAddSingleton(sp => sp.GetRequiredService<IFactory<FhirClient>>().Create());
+            var url = new Uri(Environment.GetEnvironmentVariable("FhirService:Url"));
```

Comment:
You can use the Configuration object versus calling Environment.GetEnviornmentVariable. This allows you to read configuration values from any/all registered providers (env, from a config file, etc)

Reply:
Updated to use the configuration object. Also moved code out to extension method